File: 크레인인형뽑기게임/javascript/20221104/solution.test.js
import solution from './solution';

test('solution', () => {
  const moves = [1, 5, 3, 5, 1, 2, 1, 4];
  const board = [[0, 0, 0, 0, 0],
    [0, 0, 1, 0, 3],
    [0, 2, 5, 0, 1],
    [4, 2, 4, 4, 2],
    [3, 5, 1, 3, 1]];
  expect(solution(board, moves)).toStrictEqual(4);
});

File: programmers/level1/크레인인형뽑기게임/java/20221104/test/SolutionTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class SolutionTest {
    @Test
    void answer() {
        Solution solution = new Solution();

        int[] moves = {1, 5, 3, 5, 1, 2, 1, 4};
        int[][] array = {{0, 0, 0, 0, 0}, {0, 0, 1, 0, 3}, {0, 2, 5, 0, 1}, {4, 2, 4, 4, 2}, {3, 5, 1, 3, 1}};
        assertEquals(4, solution.solution(array, moves));
    }

}
